server/pull_request.go

```diff
@@ -14,16 +14,52 @@ import (
 	"github.com/mattermost/mattermost-server/mlog"
 )
 
-func handlePullRequestEvent(event *PullRequestEvent) {
-	pr, err := GetPullRequestFromGithub(event.PullRequest)
+func (s *Server) handlePullRequestEvent(event *PullRequestEvent) {
+	mlog.Info("PR-Event", mlog.String("repo", *event.Repo.Name), mlog.Int("pr", event.PRNumber), mlog.String("action", event.Action))
+	pr, err := s.GetPullRequestFromGithub(event.PullRequest)
 	if err != nil {
 		mlog.Error("Unable to get PR from GitHub", mlog.Int("pr", event.PRNumber), mlog.Err(err))
 		return
 	}
 
-	if event.Action == "closed" {
-		go checkIfNeedCherryPick(pr)
-		if result := <-Srv.Store.Spinmint().Get(pr.Number, pr.RepoName); result.Err != nil {
+	switch event.Action {
+	case "labeled":
+		if event.Label == nil {
+			mlog.Error("Label event received, but label object was empty")
+			return
+		}
+		if s.isSpinWickLabel(*event.Label.Name) {
```

Comment:
dont we need to add the old test servers as well?

Reply:
As we discussed in Mattermost, we will need to cleanup the mattermod DB after this is merged, but that can be done anytime without issue.